### ISSUE TITLE
Generate the script for all migrations that have not yet been applied to a given database

### ISSUE THREAD

**Issue by AlissonRS** [dotnet/efcore#21995]
Often we add a new migration and want to generate an update script just for what changed in this migration. This requires looking inside the migrations folder for the last migration file, copy its name, and use it like this:

`Script-Migration -From 20200804000349_PreviousMigration`

It would be really nice if there was an option to just generate a script to the latest migration:

`Script-Migration -Last`

**Comment by bricelam** [comment]
After #14122, you could do this:
```ps1
Script-Migration -From (Get-Migration | select -Last 2)[0].id
```

**Comment by ajcvickers** [comment]
@AlissonRS Can we dig a bit deeper into why you want specifically the last migration? Would it not be more useful to be able to generate the script for all migrations that have not yet been applied to a given database? This might be only one, if all the others have been applied, but it would also handle the case where multiple migrations need to be applied.

**Comment by bricelam** [comment]
The script for that would be:

``` ps1
Script-Migration -From (Get-Migration | where applied | select -Last 1).id
```

But it also probably warrants its own sugar. Something like `Script-Migration -FromLastApplied` with an optional `-ConnectionString` parameter.

**Comment by AlissonRS** [comment]
@ajcvickers generating script for all pending migrations would be even better, but I'm not sure if it's possible to detect what migrations have not been applied yet without touching DB. If I'm not mistaken `Script-Migration` doesn't touch DB right?

**Comment by bricelam** [comment]
> Script-Migration doesn't touch DB right?

Yes, this is the current design. But I think we can evolve this with an explicit user action to access the database (e.g. specifying a flag like `-FromLastApplied`)

**Comment by AlissonRS** [comment]
@bricelam sorry for such a long time without a reply. That flag sounds great. So what about two flags then?

`Script-Migration -Last` << generates script based on last migration from "Migrations" folder (no database involved)
`Script-Migration -FromLastApplied` << generates script for all pending migrations based on database

I insist on the `-Last` flag because not everyone has access to the database.

**Comment by AlissonRS** [comment]
For clarification, the scenario I originally proposed is this:

Imagine I have some existing migrations:

![image](https://user-images.githubusercontent.com/4753901/95367677-68772280-08ab-11eb-94ce-5397f11600a0.png)

Then I change my entities, now I add a new migration:

`Add-Migration SomeNewMigration`

![image](https://user-images.githubusercontent.com/4753901/95367889-b68c2600-08ab-11eb-8ee8-33fa23b20a7b.png)

If now I want a script with that change, I'd need to provide the name of the previous migration:

`Script-Migration -From 20200915210701_addMatchShot`

The idea is a flag that allows omitting that, so providing a flag `-Last` could be enough:

`Script-Migration -Last`

This shouldn't need to touch a database.

**Comment by bricelam** [comment]
Duplicate of #15762
